playbooks/installer/ospd/post.yml

```diff
@@ -19,18 +23,59 @@
 
       - block:
           - name: update file permissions
-            become: no
             file:
                 path: "{{ inventory_dir }}/id_rsa_overcloud"
                 mode: 0600
 
           - name: Update Inventory
-            become: no
             add_host:
                 name: "{{ item }}"
-                ansible_user: "heat-admin"
+                ansible_user: "{{ user }}"
             with_items: "{{ groups['overcloud_nodes'] }}"
+
+          - shell: "cat keystonerc | grep AUTH | cut -d '=' -f 2"
+            register: oc_auth
+
+          - shell: "cat keystonerc | grep USERNAME | cut -d '=' -f 2"
+            register: oc_username
+
+          - shell: "cat keystonerc | grep TENANT | cut -d '=' -f 2"
+            register: oc_tenant
+
+          - shell: "cat keystonerc | grep PASSWORD | cut -d '=' -f 2"
+            register: oc_password
+
         delegate_to: localhost
+        become: no
+
+      - name: Dirty debug - install shade by hand
```

Comment:
IIUC, this will install shade on system instead of virtualenv.

Please look into Ansible's [pip](http://docs.ansible.com/ansible/pip_module.html) module as it can also create virtualenv